src/EditorFeatures/CSharpTest/UseNullPropagation/UseNullPropagationTests.cs

```diff
@@ -360,5 +360,75 @@ public void Method<T>(Expression<Func<T, string>> functor)
     }
 }");
         }
+
+        [WorkItem(19774, "https://github.com/dotnet/roslyn/issues/19774")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseNullPropagation)]
+        public async Task TestNullableMemberAccess()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+
+class C
+{
+    void Main(DateTime? toDate)
+    {
+        var v = [||]toDate == null ? null : toDate.Value.ToString(""yyyy/MM/ dd"");
+    }
+}
+",
+
+@"
+using System;
+
+class C
+{
+    void Main(DateTime? toDate)
+    {
+        var v = toDate?.ToString(""yyyy/MM/ dd"");
+    }
+}
+");
+        }
+
+        [WorkItem(19774, "https://github.com/dotnet/roslyn/issues/19774")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseNullPropagation)]
+        public async Task TestNullableElementAccess()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+
+struct S
+{
+    public string this[int i] => """";
+}
+
+class C
+{
+    void Main(S? s)
+    {
+        var x = [||]s == null ? null : s.Value[0];
```

Comment:
:bulb: It would be good to add a test which triggers both sides of the code fix logic, such as the following:

```csharp
void Main(bool? s)
{
  var x = s == null ? null : s.Value.ToString() + s.HasValue.ToString();
}
```